apps/web/components/eventtype/RecurringEventController.tsx

```diff
@@ -86,36 +74,37 @@ export default function RecurringEventController({
                 </label>
               </div>
             </div>
-            <Collapsible
-              open={recurringEventDefined}
-              data-testid="recurring-event-collapsible"
-              onOpenChange={() => setRecurringEventDefined(!recurringEventDefined)}>
-              <CollapsibleContent className="mt-4 text-sm">
+            {recurringEventState && (
+              <div data-testid="recurring-event-collapsible" className="mt-4 text-sm">
                 <div className="flex items-center">
                   <p className="mr-2 text-neutral-900">{t("repeats_every")}</p>
                   <input
                     type="number"
                     min="1"
                     max="20"
                     className="block w-16 rounded-sm border-gray-300 shadow-sm [appearance:textfield] ltr:mr-2 rtl:ml-2 sm:text-sm"
-                    defaultValue={recurringEvent?.interval || 1}
+                    defaultValue={recurringEventState.interval}
                     onChange={(event) => {
-                      setRecurringEventInterval(parseInt(event?.target.value));
-                      recurringEvent.interval = parseInt(event?.target.value);
-                      formMethods.setValue("recurringEvent", recurringEvent);
+                      const newVal = {
+                        ...recurringEventState,
+                        interval: parseInt(event?.target.value),
+                      };
+                      formMethods.setValue("recurringEvent", newVal);
```

Comment:
`Controller` can be used here to simplify, but not needed - this is already a great PR.